plugins/modules/provision_workflow_manager.py

```diff
@@ -1046,77 +1151,54 @@ def initialize_wired_provisioning(self):
-            self.result['msg'] = self.msg
-            self.result['response'] = execution_id
-            self.log(self.result['msg'], "INFO")
-            return self
-        except Exception as e:
-            self.msg = "Error in site assignment: {0}".format(str(e))
-            self.log(self.msg, "ERROR")
-            self.status = "failed"
-            self.result['response'] = self.msg
-            self.check_return_status()
+        if self.compare_dnac_versions(self.get_ccc_version(), "2.3.5.3") <= 0:
+            try:
+                response = self.dnac_apply['exec'](
+                    family="sda",
+                    function="provision_wired_device",
+                    op_modifies=True,
+                    params=self.want["prov_params"],
+                )
+                if response:
+                    self.log("Received API response from 'provision_wired_device': {0}".format(str(response)), "DEBUG")
+                    if self.status not in ["failed", "exited"]:
+                        self.log("Wired Device '{0}' provisioning completed successfully.".format(self.device_ip), "INFO")
+                        self.result["changed"] = True
+                        self.result['msg'] = "Provisioning of the device '{0}' completed successfully.".format(self.device_ip)
+                        self.result['response'] = "Provisioning of the device '{0}' completed successfully.".format(self.device_ip)
+                        self.log(self.result['msg'], "INFO")
+                        return self
+
+            except Exception as e:
+                self.msg = "Error in provisioning device '{0}' due to {1}".format(self.device_ip, str(e))
+                self.log(self.msg, "ERROR")
+                self.status = "failed"
+                self.result['response'] = self.msg
+                self.check_return_status()
+        else:
+            try:
+                response = self.dnac._exec(family="sda", function='provision_devices', op_modifies=True, params={"payload": provision_params})
+                if response:
+                    self.log("Received API response from 'provision_devices': {0}".format(str(response)), "DEBUG")
+                    self.check_tasks_response_status(response, api_name='provision_device')
+                    if self.status not in ["failed", "exited"]:
+                        self.log("Wired Device '{0}' provisioning completed successfully.".format(self.device_ip), "INFO")
+                        self.result["changed"] = True
+                        self.result['msg'] = "Provisioning of the device '{0}' completed successfully.".format(self.device_ip)
+                        self.result['response'] = "Provisioning of the device '{0}' completed successfully.".format(self.device_ip)
+                        self.log(self.result['msg'], "INFO")
+                        return self
+
+                if self.status not in ["failed", "exited"]:
```

Comment:
what is the use of it?

Reply:
Updated the code accordingly. This If statement is not needed.